packages/jest-snapshot/src/__tests__/throwMatcher.test.ts

```diff
@@ -5,59 +5,65 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import {toThrowErrorMatchingSnapshot} from '..';
+import {type Context, toThrowErrorMatchingSnapshot} from '../';
 
-let matchFn: jest.Mock;
+const mockedMatch = jest.fn(() => ({
+  actual: 'coconut',
+  expected: 'coconut',
+}));
 
-beforeEach(() => {
-  matchFn = jest.fn(() => ({
-    actual: 'coconut',
-    expected: 'coconut',
-  }));
+const mockedContext = {
+  snapshotState: {match: mockedMatch},
+} as unknown as Context;
+
+afterEach(() => {
+  jest.clearAllMocks();
 });
 
 it('throw matcher can take func', () => {
-  const throwMatcher = toThrowErrorMatchingSnapshot.bind({
-    snapshotState: {match: matchFn},
-  });
-
-  throwMatcher(
+  toThrowErrorMatchingSnapshot.call(
+    mockedContext,
     () => {
       throw new Error('coconut');
     },
     undefined,
     false,
   );
 
-  expect(matchFn).toHaveBeenCalledWith(
+  expect(mockedMatch).toBeCalledTimes(1);
+  expect(mockedMatch).toHaveBeenCalledWith(
     expect.objectContaining({received: 'coconut', testName: ''}),
   );
 });
 
 describe('throw matcher from promise', () => {
-  let throwMatcher: typeof toThrowErrorMatchingSnapshot;
-
-  beforeEach(() => {
-    throwMatcher = toThrowErrorMatchingSnapshot.bind({
-      snapshotState: {match: matchFn},
-    });
-  });
```

Comment:
`.bind` was reworked into `.call`, which proofed to be friendlier with types. Also mocks are cleared in `afterEach` calling `jest.clearAllMocks()`.